### PR TITLE
trace: Enable DB statement obfuscation in span tags by default

### PR DESCRIPTION
**Pull request by katiehockman**
### What does this PR do?

Enable DB statement obfuscation for span tags by default which was previously off by default.
This affects MongoDB queries, ElasticSearch request bodies, and raw commands from Redis and MemCached

This change does not affect resource name obfuscation, as that has been and still is on by default.

### Motivation

Improve the default customer experience regarding sensitive data.

### Possible Drawbacks / Trade-offs

This change could cause a significant increase in CPU or memory usage, or could obfuscate
data that is not sensitive which someone wishes to keep un-obfuscated.
To mitigate this, obfuscation rules can still be fine-tuned or disabled with the Agent config.

### Describe how to test/QA your changes

Create an instrumented query from one of the frameworks listed above which includes sensitive data,
and verify that this data is obfuscated by the default Agent config.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by ajgajg1134 — APPROVED** [review]

File: releasenotes/notes/apm-obfuscation-on-by-default-1ccc2e32cdd10ea4.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+#
+# Each section note must be formatted as reStructuredText.
+---
+security:
+  - |
+    APM: In order to improve the default customer experience regarding
+    sensitive data, the Agent now obfuscates database statements within
+    span metadata by default. This includes MongoDB queries,
+    ElasticSearch request bodies, and raw commands from Redis and
+    MemCached. This was off by default, previously.
```
Comment (rtrieu) [inline releasenotes/notes/apm-obfuscation-on-by-default-1ccc2e32cdd10ea4.yaml:15]:
```suggestion
    MemCached. Previously, this setting was off by default.
```

File: releasenotes/notes/apm-obfuscation-on-by-default-1ccc2e32cdd10ea4.yaml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    This could have performance implications, or obfuscate data that
```
Comment (rtrieu) [inline releasenotes/notes/apm-obfuscation-on-by-default-1ccc2e32cdd10ea4.yaml:16]:
```suggestion
    This update could have performance implications, or obfuscate data that
```

File: releasenotes/notes/apm-obfuscation-on-by-default-1ccc2e32cdd10ea4.yaml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    is not sensitive, and can be disabled or configured via the
```
Comment (rtrieu) [inline releasenotes/notes/apm-obfuscation-on-by-default-1ccc2e32cdd10ea4.yaml:17]:
```suggestion
    is not sensitive, and can be disabled or configured through the
```

File: releasenotes/notes/apm-obfuscation-on-by-default-1ccc2e32cdd10ea4.yaml [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+    obfuscation options within the apm_config, or with the
+    environment variables prefixed with DD_APM_OBFUSCATION. Please read the
```
Comment (rtrieu) [inline releasenotes/notes/apm-obfuscation-on-by-default-1ccc2e32cdd10ea4.yaml:19]:
```suggestion
    environment variables prefixed with `DD_APM_OBFUSCATION`. Please read the
```

File: releasenotes/notes/apm-obfuscation-on-by-default-1ccc2e32cdd10ea4.yaml [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+security:
+  - |
+    APM: In order to improve the default customer experience regarding
+    sensitive data, the Agent now obfuscates database statements within
+    span metadata by default. This includes MongoDB queries,
+    ElasticSearch request bodies, and raw commands from Redis and
+    MemCached. This was off by default, previously.
+    This could have performance implications, or obfuscate data that
+    is not sensitive, and can be disabled or configured via the
+    obfuscation options within the apm_config, or with the
```
Comment (rtrieu) [inline releasenotes/notes/apm-obfuscation-on-by-default-1ccc2e32cdd10ea4.yaml:18]:
```suggestion
    obfuscation options within the `apm_config`, or with the
```

**Review by rtrieu — CHANGES_REQUESTED** [review]
A few suggestions, but otherwise looks good!

**Review by ogaca-dd — APPROVED** [review]

**Review by rtrieu — APPROVED** [review]

**Pull request merged**